sources/packages/api/src/esdc-integration/e-cert-integration/e-cert-part-time-integration/e-cert-files/e-cert-file-record.ts

```diff
@@ -0,0 +1,193 @@
+    const record = new StringBuilder();
+    record.append(this.recordType);
+    record.appendWithEndFiller(this.lastName, 25, SPACE_FILLER);
+    record.appendWithEndFiller(this.firstName || "", 15, SPACE_FILLER);
+    record.repeatAppend(SPACE_FILLER, 3); // Initials, optional, not provided.
+    record.append(this.sin, 9);
+    record.append(this.gender, 1);
+    record.appendDate(this.dateOfBirth, DATE_FORMAT);
+    record.appendWithEndFiller(this.maritalStatus, 4, SPACE_FILLER);
+    record.appendWithEndFiller(this.addressLine1, 40, SPACE_FILLER);
+    record.appendWithEndFiller(this.addressLine2 || "", 40, SPACE_FILLER);
+    record.repeatAppend(SPACE_FILLER, 40); // AddressLine 3, optional, not provided.
+    record.appendWithEndFiller(this.city, 25, SPACE_FILLER);
+    record.append("BC  ", 4); //TODO Province, is hardcoded to "BC  ".
+    record.append("CAN "); // TODO Country, is hardcoded to "CAN ".
+    record.repeatAppend(SPACE_FILLER, 16); //TODO Postal code, Filled with space as not provided.
+    record.repeatAppend(SPACE_FILLER, 16); // Telephone, optional, not provided.
+    record.repeatAppend(SPACE_FILLER, 40); // Alternate Address 1, optional, not provided.
+    record.repeatAppend(SPACE_FILLER, 40); // Alternate Address 2, optional, not provided.
+    record.repeatAppend(SPACE_FILLER, 40); // Alternate Address 3, optional, not provided.
+    record.repeatAppend(SPACE_FILLER, 25); // Alternate City, optional, not provided.
+    record.repeatAppend(SPACE_FILLER, 4); // Alternate Province, optional, not provided.
+    record.repeatAppend(SPACE_FILLER, 4); // Alternate Country, optional, not provided.
+    record.repeatAppend(SPACE_FILLER, 16); // Alternate Postal Code, optional, not provided.
+    record.repeatAppend(SPACE_FILLER, 16); // Alternate Telephone, optional, not provided.
+    record.appendWithEndFiller(this.studentNumber || "", 12, SPACE_FILLER);
+    record.appendDate(this.disbursementDate, DATE_FORMAT);
+    record.appendWithStartFiller(this.disbursementAmount, 9, NUMBER_FILLER);
+    record.appendWithStartFiller(this.certNumber, 7, NUMBER_FILLER);
+    record.appendDate(this.educationalStartDate, DATE_FORMAT);
+    record.appendDate(this.educationalEndDate, DATE_FORMAT);
+    record.appendWithEndFiller(this.federalInstitutionCode, 4, SPACE_FILLER);
+    record.appendWithStartFiller(this.fieldOfStudy, 4, NUMBER_FILLER);
+    record.append(this.yearOfStudy.toString(), 1);
+    record.appendWithStartFiller(this.weeksOfStudy, 3, NUMBER_FILLER);
+    record.appendDate(this.documentProducedDate, DATE_FORMAT);
+    record.repeatAppend(SPACE_FILLER, 8); // TODO Cancel Date, E-cert cancellation date.
+    record.repeatAppend(SPACE_FILLER, 9); //CAG PD Amt, No longer used.
+    record.repeatAppend(SPACE_FILLER, 9); //CAG LI Amt, No longer used.
+    record.appendWithStartFiller(this.totalGrantAmount, 5, NUMBER_FILLER); //Sum of all CSGP grants + BC Provincial Part-time grants from this certificate.
```

Comment:
I would recommend moving these comments to the properties. If they are already there, there is no need to keep them here also.

Reply:
This comment is for the developer to have an idea of what is given in the documentation that the ministry provided, without opening them 😉

Reply:
As we talked, it could be added to the property to be clear to the person consuming this class also 😉